### PR TITLE
Set segment pill button corner radius

### PR DESCRIPTION
**Pull request by joannaquu**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

Currently, `SegmentedControl` sets the `cornerRadius`. On iOS, this trickles down to `SegmentPillButton` as well. This is not true on other platforms. To fix this, set the `cornerRadius` explicitly.

To avoid magic numbers, move `SegmentedControl.Constants` to `SegmentedControlTokenSet` and access the corner radius from there. While I'm here, remove `selectionBarHeight` since it's not being used anywhere.

### Binary change

Total increase: 2,224 bytes
Total decrease: -1,600 bytes
| File | Before | After | Delta |
|------|-------:|------:|------:|
| Total | 30,917,640 bytes | 30,918,264 bytes | ⚠️ 624 bytes |
<details>
<summary> Full breakdown </summary>

| File | Before | After | Delta |
|------|-------:|------:|------:|
| SegmentedControlTokenSet.o | 90,128 bytes | 91,448 bytes | ⚠️ 1,320 bytes |
| SegmentPillButton.o | 90,648 bytes | 91,552 bytes | ⚠️ 904 bytes |
| FocusRingView.o | 814,848 bytes | 814,688 bytes | 🎉 -160 bytes |
| __.SYMDEF | 4,770,960 bytes | 4,770,720 bytes | 🎉 -240 bytes |
| SegmentedControl.o | 333,384 bytes | 332,184 bytes | 🎉 -1,200 bytes |
</details>

### Verification

<details>
<summary>Visual Verification</summary>

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| before | after |
</details>

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/fluentui-apple/pull/1963)

### CODE REVIEW

**Review by laminesm — APPROVED** [review]
Should we consider making this a constant token instead of using a magic number?

File: ios/FluentUI/SegmentedControl/SegmentPillButton.swift [hunk 1]
```diff
@@ -41,6 +41,7 @@ class SegmentPillButton: UIButton {
                                                               trailing: horizontalInset)
         configuration.background.backgroundColor = .clear
         configuration.baseForegroundColor = tokenSet[.restLabelColor].uiColor
+        configuration.background.cornerRadius = 16
```
Comment (mischreiber) [inline ios/FluentUI/SegmentedControl/SegmentPillButton.swift:44]: As mentioned by @laminesm, I'd like to see this stored as a named constant somewhere--either as a token constant, or potentially added as a `CornerRadiusToken.radius160`, though that would require updating the token design library.

Comment (mischreiber) [reply ios/FluentUI/SegmentedControl/SegmentPillButton.swift:44]: Alternatively, can we just use `CornerRadiusToken.radiusCircular` for the same effect?

Comment (joannaquu) [reply ios/FluentUI/SegmentedControl/SegmentPillButton.swift]: `CornerRadiusToken.radiusCircular` would result in a corner radius that's too large. It's already stored in `SegmentedControl.Constants` so I'll move that to `SegmentedControlTokenSet` and access it there.

**Review by mischreiber — APPROVED** [review]

**Pull request merged**